File: 3047-maximum-element-sum-of-a-complete-subset-of-indices/maximum-element-sum-of-a-complete-subset-of-indices.java
class Solution {
    public long maximumSum(List<Integer> nums) {
        long res = 0;
        for (int k = 1; k <= nums.size(); k++) {
            long cur = 0;
            for (int v = 1; v * v * k <= nums.size(); v++)
                cur += nums.get(k * v * v - 1);
            res = Math.max(res, cur);
        }
        return res;
    }
}

File: 3047-maximum-element-sum-of-a-complete-subset-of-indices/maximum-element-sum-of-a-complete-subset-of-indices.py
class Solution:
    def maximumSum(self, nums: List[int]) -> int:
        count = Counter()
        for i in range(len(nums)):
            x, v = i + 1, 2
            while x >= v * v:
                while x % (v * v) == 0:
                    x //= v * v
                v += 1
            count[x] += nums[i]
        return max(count.values())